internal/tui/runner.go

```diff
@@ -0,0 +1,362 @@
+	),
+	Up: key.NewBinding(
+		key.WithKeys("up", "k"),
+		key.WithHelp("↑/k", "up"),
+	),
+	Down: key.NewBinding(
+		key.WithKeys("down", "j"),
+		key.WithHelp("↓/j", "down"),
+	),
+	Space: key.NewBinding(
+		key.WithKeys(" "),
+		key.WithHelp("space", "select"),
+	),
+}
+
+// questionnaireModel handles multiple questions
+type questionnaireModel struct {
+	questions       []*Question
+	currentQuestion int
+	answers         map[string]interface{}
+	finished        bool
+	err             error
+	width           int
+	height          int
+	help            help.Model
+}
+
+func newQuestionnaireModel(questions []*Question) *questionnaireModel {
+	return &questionnaireModel{
+		questions: questions,
+		answers:   make(map[string]interface{}),
+		width:     80,
+		height:    24,
+		help:      help.New(),
+	}
+}
+
+func (m *questionnaireModel) Init() tea.Cmd {
+	return nil
+}
+
+func (m *questionnaireModel) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
+	switch msg := msg.(type) {
+	case tea.WindowSizeMsg:
+		m.width = msg.Width
+		m.height = msg.Height
+		return m, nil
+
+	case tea.KeyMsg:
+		if m.finished {
+			return m, tea.Quit
+		}
+
+		switch msg.String() {
+		case "ctrl+c":
+			m.err = fmt.Errorf("cancelled by user")
+			return m, tea.Quit
+
+		case "enter":
+			if err := m.validateCurrentAnswer(); err != nil {
+				m.setCurrentError(err.Error())
+				return m, nil
+			}
+
+			// Save current answer
+			current := m.questions[m.currentQuestion]
+			m.answers[current.Name] = current.Prompt.Value()
+
+			// Move to next question or finish
+			m.currentQuestion++
+			if m.currentQuestion >= len(m.questions) {
+				m.finished = true
+				return m, tea.Quit
+			}
+
+			// Clear any previous error
+			m.setCurrentError("")
+			return m, nil
+		}
+	}
+
+	// Update current prompt
+	if !m.finished && m.currentQuestion < len(m.questions) {
+		current := m.questions[m.currentQuestion]
+		updatedPrompt, cmd := current.Prompt.Update(msg)
+		current.Prompt = updatedPrompt
+		return m, cmd
+	}
+
+	return m, nil
+}
+
+func (m *questionnaireModel) validateCurrentAnswer() error {
+	current := m.questions[m.currentQuestion]
+	if current.Validate != nil {
+		return current.Validate(current.Prompt.Value())
+	}
+	return nil
+}
+
+func (m *questionnaireModel) setCurrentError(err string) {
+	if m.currentQuestion < len(m.questions) {
+		current := m.questions[m.currentQuestion]
+		switch p := current.Prompt.(type) {
+		case interface{ SetError(string) }:
+			p.SetError(err)
+		}
+	}
+}
+
+func (m *questionnaireModel) View() string {
+	if m.finished {
+		// When finished, show the final summary of all answers
+		return m.renderFinalSummary()
+	}
+
+	var b strings.Builder
+
+	// Display previous questions and their answers
+	for i := 0; i < m.currentQuestion; i++ {
+		question := m.questions[i]
+		if answer, exists := m.answers[question.Name]; exists {
+			answerStr := m.formatAnswer(answer)
+			questionLine := fmt.Sprintf("? %s: %s", question.Prompt.Message(), answerStr)
```

Comment:
Nit. If we want to keep the previous style, the `?` should be shown in a different color.